diff -Naur ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativecomponent.cpp ./qdeclarativecomponent.cpp
--- ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativecomponent.cpp	2013-06-18 15:44:20.271297040 -0700
+++ ./qdeclarativecomponent.cpp	2013-06-20 16:35:50.675276052 -0700
@@ -800,6 +800,7 @@
         QDeclarativeData *ddata = QDeclarativeData::get(rv);
         Q_ASSERT(ddata);
         ddata->indestructible = true;
+        ddata->inCreation = false;
     }
     return rv;
 }
diff -Naur ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativedata_p.h ./qdeclarativedata_p.h
--- ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativedata_p.h	2012-03-14 07:01:13.000000000 -0700
+++ ./qdeclarativedata_p.h	2013-06-20 16:18:00.035276180 -0700
@@ -74,7 +74,7 @@
 {
 public:
     QDeclarativeData()
-        : ownMemory(true), ownContext(false), indestructible(true), explicitIndestructibleSet(false), 
+        : ownMemory(true), ownContext(false), indestructible(true), explicitIndestructibleSet(false), inCreation(false),
           context(0), outerContext(0), bindings(0), nextContextObject(0), prevContextObject(0), bindingBitsSize(0), 
           bindingBits(0), lineNumber(0), columnNumber(0), deferredComponent(0), deferredIdx(0), 
           scriptValue(0), objectDataRefCount(0), propertyCache(0), guards(0), extendedData(0) {
@@ -103,6 +103,7 @@
     quint32 ownContext:1;
     quint32 indestructible:1;
     quint32 explicitIndestructibleSet:1;
+    quint32 inCreation:1;
     quint32 dummy:28;
 
     // The context that created the C++ object
diff -Naur ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativeobjectscriptclass.cpp ./qdeclarativeobjectscriptclass.cpp
--- ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativeobjectscriptclass.cpp	2012-03-14 07:01:13.000000000 -0700
+++ ./qdeclarativeobjectscriptclass.cpp	2013-06-20 16:45:29.375275985 -0700
@@ -77,7 +77,7 @@
     virtual ~ObjectData() {
         if (object && !object->parent()) {
             QDeclarativeData *ddata = QDeclarativeData::get(object, false);
-            if (ddata && !ddata->indestructible && 0 == --ddata->objectDataRefCount) 
+            if (ddata && !ddata->indestructible && !ddata->inCreation && 0 == --ddata->objectDataRefCount)
                 object->deleteLater();
         }
     }
diff -Naur ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativevme.cpp ./qdeclarativevme.cpp
--- ../../../../../qt-everywhere-opensource-src-4.8.1/src/declarative/qml//qdeclarativevme.cpp	2012-03-14 07:01:13.000000000 -0700
+++ ./qdeclarativevme.cpp	2013-06-20 16:50:50.291275944 -0700
@@ -234,6 +234,8 @@
                 QDeclarativeData *ddata = QDeclarativeData::get(o);
                 Q_ASSERT(ddata);
 
+                ddata->inCreation = true;
+
                 if (stack.isEmpty()) {
                     if (ddata->context) {
                         Q_ASSERT(ddata->context != ctxt);
